improve(finalizer): remove hardcoded constants

bmzig opened the pull request:
In preparation for NCA, we should remove all instances where we hardcode the chain IDs so that the bots will not automatically assume we are on mainnet.

pxrl commented on src/dataworker/Dataworker.ts:
```diff
@@ -1262,7 +1263,7 @@ export class Dataworker {
           // If polygon, keep separate from relayer refund leaves since we can't execute refunds atomically
           // with fills.
-          groupId: chainId === 137 ? "slowRelay" : undefined,
+          groupId: chainId === CHAIN_IDs.POLYGON ? "slowRelay" : undefined,
```
We probably want to use `chainIsMatic()` here instead, since that should also handle Polygon PoS testnets.

https://github.com/across-protocol/sdk/blob/8e32dab62a0f9f72e42b327921b64fdfa71b34ea/src/utils/NetworkUtils.ts#L50

bmzig replied:
Fixed: https://github.com/across-protocol/relayer/pull/1610/commits/aa75abdc59ed33b374f07c3caa0d80e6b3e2ed8c

pxrl commented on src/finalizer/index.ts:
```diff
@@ -59,55 +60,59 @@ type FinalizationType = "l1->l2" | "l2->l1" | "l1<->l2";
+  [CHAIN_IDs.LINEA_GOERLI]: {
     finalizeOnL1: [lineaL2ToL1Finalizer],
     finalizeOnL2: [lineaL1ToL2Finalizer],
   },
```
Linea have a Sepolia network now, so we can probably circle back afterwards to remove this.

bmzig replied:
Fixed: https://github.com/across-protocol/relayer/pull/1610/commits/aa75abdc59ed33b374f07c3caa0d80e6b3e2ed8c

pxrl commented on src/finalizer/index.ts:
```diff
@@ -59,55 +60,59 @@ type FinalizationType = "l1->l2" | "l2->l1" | "l1<->l2";
     finalizeOnL1: [opStackFinalizer],
     finalizeOnL2: [],
   },
+  [CHAIN_IDs.ZK_SYNC_GOERLI]: {
+    finalizeOnL1: [zkSyncFinalizer],
+    finalizeOnL2: [],
+  },
```
Should be fine to drop this too - we don't use it.

bmzig replied:
Fixed: https://github.com/across-protocol/relayer/pull/1610/commits/aa75abdc59ed33b374f07c3caa0d80e6b3e2ed8c

pxrl commented on yarn.lock:
Curious about this - did we get out of sync? :thinking:

bmzig replied:
Does using a different node version change the yarn.lock? I'm using ver. 18

bmzig replied:
How did you end up fixing it? This seems to happen for all of my commits.

pxrl reviewed (commented):
Looks good - couple of small comments.

pxrl commented:
@bmzig Looking good! I see some additional hardcoding in:
- `src/finalizer/utils/opStack.ts`
- `src/clients/bridges/ArbitrumAdapter.ts`
- `src/clients/bridges/op-stack/optimism/OptimismAdapter.ts`
- `src/clients/bridges/BaseAdapter.ts`
- `src/dataworker/Dataworker.ts`: `unpermissioned: Number(chainId) === 1`

bmzig commented:
> @bmzig Looking good! I see some additional hardcoding in:
> 
> * `src/finalizer/utils/opStack.ts`
> * `src/clients/bridges/ArbitrumAdapter.ts`
> * `src/clients/bridges/op-stack/optimism/OptimismAdapter.ts`
> * `src/clients/bridges/BaseAdapter.ts`
> * `src/dataworker/Dataworker.ts`: `unpermissioned: Number(chainId) === 1`
> 
> (I also found a few more and dropped them in this PR: #1611)



> @bmzig Looking good! I see some additional hardcoding in:
> 
> * `src/finalizer/utils/opStack.ts`
> * `src/clients/bridges/ArbitrumAdapter.ts`
> * `src/clients/bridges/op-stack/optimism/OptimismAdapter.ts`
> * `src/clients/bridges/BaseAdapter.ts`
> * `src/dataworker/Dataworker.ts`: `unpermissioned: Number(chainId) === 1`
> 
> (I also found a few more and dropped them in this PR: #1611)



> @bmzig Looking good! I see some additional hardcoding in:
> 
> * `src/finalizer/utils/opStack.ts`
> * `src/clients/bridges/ArbitrumAdapter.ts`
> * `src/clients/bridges/op-stack/optimism/OptimismAdapter.ts`
> * `src/clients/bridges/BaseAdapter.ts`
> * `src/dataworker/Dataworker.ts`: `unpermissioned: Number(chainId) === 1`
> 
> (I also found a few more and dropped them in this PR: #1611)



> @bmzig Looking good! I see some additional hardcoding in:
> 
> * `src/finalizer/utils/opStack.ts`
> * `src/clients/bridges/ArbitrumAdapter.ts`
> * `src/clients/bridges/op-stack/optimism/OptimismAdapter.ts`
> * `src/clients/bridges/BaseAdapter.ts`
> * `src/dataworker/Dataworker.ts`: `unpermissioned: Number(chainId) === 1`
> 
> (I also found a few more and dropped them in this PR: #1611)

The top/bottom are fixed here: https://github.com/across-protocol/relayer/pull/1610/commits/a584a4b0e02d29a8344f5522e2824a429cc4d32d

I'm saving all the hardcoded values in bridges/* for https://github.com/across-protocol/relayer/pull/1524, since I'm aiming to lean down the code/make those instances of hardcoded values generic.

pxrl reviewed (commented):
Just found that I had these comments pending. I'll circle over the PR again now.

pxrl commented on src/finalizer/utils/opStack.ts:
```diff
@@ -109,10 +111,6 @@ export async function opStackFinalizer(
-
 function getOptimismClient(chainId: OVM_CHAIN_ID, hubSigner: Signer): OVM_CROSS_CHAIN_MESSENGER {
   return new optimismSDK.CrossChainMessenger({
     bedrock: true,
```
I spotted a sly `l1ChainId: 1` on the next line, as well as on line 119 (`l1SignerOrProvider: ...(1, true)),`). This stuff is pervacious :(

bmzig replied:
Got it here: https://github.com/across-protocol/relayer/pull/1610/commits/7d36ef2dfdf0bef8886808e7eea8c680e78e13de

pxrl commented on src/finalizer/utils/opStack.ts:
```diff
@@ -34,7 +34,9 @@ interface CrossChainMessageWithStatus extends CrossChainMessageWithEvent {
-type OVM_CHAIN_ID = 10 | 8453 | 34443;
+const OP_STACK_CHAINS = Object.values(CHAIN_IDs).filter((chainId) => chainIsOPStack(chainId));
+
+type OVM_CHAIN_ID = (typeof OP_STACK_CHAINS)[number];
```
I'm not 100% across what this is doing under the hood...but I think it'd probably also be fine if we just changed all instance of `OVM_CHAIN_ID` to `number`. The function `opStackFinalizer` is the sole entry point and it already asserts that the input chain ID maps to an OP stack.

bmzig replied:
I added a comment since when we use this syntax, in case we do want this type in the future it will not bug out. Although, if it seems a bit too convoluted, I can just make it a number. 
https://github.com/across-protocol/relayer/pull/1610/commits/7d36ef2dfdf0bef8886808e7eea8c680e78e13de

pxrl reviewed (approved):
LGTM subject to some straggling hardcoded chain IDs, and potentially revisiting the OVM_CHAIN_ID (or add a comment describing the typeof operation).
